File: Medium/785. Is Graph Bipartite/Solution.js
let ans = true;
var dfs = (graph, kind, vis, node) => {
    vis[node] = true;

    for(let child of graph[node]) {
        if(!vis[child]) {
            kind[child] = 1 - kind[node];
            dfs(graph, kind, vis, child);
        } else if(kind[child] === kind[node]) ans = false;
    }
}

var isBipartite = function(graph) {
    let n = graph.length;

    let kind = new Array(n).fill(0);
    let vis = new Array(n).fill(false);
    ans = true;

    for(let i = 0; i < n; i++)
        if(!vis[i])
            dfs(graph, kind, vis, i);

    return ans;
};

File: Medium/785. Is Graph Bipartite/Solution.java
class Solution {
    public boolean bipartite;
    public void dfs(int[][] graph, int[] kind, boolean[] vis, int node) {
        vis[node] = true;

        for(int child : graph[node]) {
            if(!vis[child]) {
                kind[child] = 1 - kind[node];
                dfs(graph, kind, vis, child);
            }
            else if(kind[child] == kind[node])
                bipartite = false;
        }
    }
    public boolean isBipartite(int[][] graph) {
        int n = graph.length;
        int[] kind = new int[n];
        boolean[] vis = new boolean[n];

        Arrays.fill(kind, 0);
        Arrays.fill(vis, false);

        bipartite = true;

        for(int i = 0; i < n; i++)
            if(!vis[i])
                dfs(graph, kind, vis, i);

        return bipartite;
    }
}